### PR TITLE
fix(ssh): add a fixed SSH key to use with `gcloud`

### LINKED ISSUE #5645: ci: ssh error after implementing OS-Login: Permission denied (publickey)
## Motivation

After merging #5602 and #5617 we're still dealing with SSH issues

### Designs

- Use a more suitable SSH mechanism for automated scripts, like using a fixed SSH key (as this might be our best workaround so far)
- Validate if we can use plain SSH with OS-login with this action https://github.com/marketplace/actions/ssh-remote-commands

### PR DESCRIPTION
**Pull request by gustavovalverde**
## Motivation

After implementing #5602 most SSH errors were reduced to a single one, but this single error was being more constant than previous ones, as this error was raised on every other workflow job

Fixes: #5645

### Designs

- Disable OS Login
- Use a fixed SSH key
- Add a public key to the metadata of the project

## Solution

- As there's no easy way to convert a secret to a file, we're using https://github.com/shimataro/ssh-key-action to accomplish this
- `gcloud compute ssh` also expects the public key to be available locally, even though it has been added to the project metadata, so we're creating a public key on the fly (from the private key)

## Review

Anyone from @ZcashFoundation/devops-reviewers 

### Reviewer Checklist

  - [x] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [x] Are the PR labels correct?
  - [x] Does the code do what the ticket and PR says?
  - [ ] How do you know it works? Does it have tests?

### CODE REVIEW

File: .github/workflows/deploy-gcp-tests.yml [hunk 1]
```diff
@@ -566,7 +605,6 @@ jobs:
       - name: Show logs for ${{ inputs.test_id }} test (sprout)
         run: |
           gcloud compute ssh ${{ inputs.test_id }}-${{ env.GITHUB_REF_SLUG_URL }}-${{ env.GITHUB_SHA_SHORT }} \
-          --ssh-key-expire-after=30s \
```
Comment (teor2345) [inline .github/workflows/deploy-gcp-tests.yml:569]: This job does not install the key, but it looks like it should.

About half the jobs in this file are missing an install step.

File: .github/workflows/deploy-gcp-tests.yml [hunk 2]
```diff
@@ -1362,15 +1434,3 @@ jobs:
           else
             gcloud compute instances delete "${INSTANCE}" --zone "${{ env.ZONE }}" --delete-disks all --quiet
           fi
-
-      # Deletes SSH keys generated during this workflow run, as GCP has a limit of SSH keys
-      # that can exist at the same time in the OS Login metadata. Not deleting this keys
-      # could cause the following error:
-      # `Login profile size exceeds 32 KiB. Delete profile values to make additional space`
-      - name: Delete temporal SSH keys
```
Comment (teor2345) [inline .github/workflows/deploy-gcp-tests.yml:1370]: Will outdated PRs continue to create these keys until they are all rebased on this PR? 
How will we do a final cleanup of these unused keys?

Do we need to keep on running this step until the last PR has been updated?

File: .github/workflows/deploy-gcp-tests.yml (hunk 1 above)
Comment (gustavovalverde) [reply .github/workflows/deploy-gcp-tests.yml:569]: Solved with [fix: add missing SSH key installation steps](https://github.com/ZcashFoundation/zebra/pull/5671/commits/81c74a292f6f936f4ec574bf40688aca9dae4be0)

File: .github/workflows/deploy-gcp-tests.yml (hunk 2 above)
Comment (gustavovalverde) [reply .github/workflows/deploy-gcp-tests.yml:1370]: This were specific keys created by the OS-Login function which was added project-wide on GCP. This is no longer happening (as it was disabled), and all unused keys have been manually deleted.

We anyways need to rebase all PRs to stop SSH errors from happening

**Review by arya2 — APPROVED** [review]
Let's try it!

File: .github/workflows/deploy-gcp-tests.yml [hunk 3]
```diff
@@ -125,6 +125,17 @@ jobs:
...
+      # Install our SSH secret
+      - name: Install private SSH key
+        uses: shimataro/ssh-key-action@v2.4.0
+        with:
+          key: ${{ secrets.GCP_SSH_PRIVATE_KEY }}
+          name: google_compute_engine
+          known_hosts: unnecessary
+
+      - name: Generate public SSH key
+        run: ssh-keygen -y -f ~/.ssh/google_compute_engine > ~/.ssh/google_compute_engine.pub
```
Comment (dconnolly) [inline .github/workflows/deploy-gcp-tests.yml:137]: Oh, so this is generated each time and pushed? OK

File: .github/workflows/deploy-gcp-tests.yml [hunk 4]
```diff
@@ -162,7 +173,6 @@ jobs:
       - name: Create ${{ inputs.test_id }} Docker volume
         run: |
           gcloud compute ssh ${{ inputs.test_id }}-${{ env.GITHUB_REF_SLUG_URL }}-${{ env.GITHUB_SHA_SHORT }} \
-          --ssh-key-expire-after=30s \
```
Comment (dconnolly) [inline .github/workflows/deploy-gcp-tests.yml:165]: 👍

**Review by dconnolly — COMMENTED** [review]
lgtm, worth a shot, ephemeral ssh keys are even better

**Pull request merged**